docs/TuxActions.md

```diff
@@ -0,0 +1,358 @@
+
+```
+    actionCategory.register(storeToRegister, actionVerbCallbackMap);
+```
+
+This method is equivalent to
+
+```
+    Actions.register(storeToRegister, {
+      todos: actionVerbCallbackMap
+    });
+```
+
+####<a id="actionCategory-ACTION"></a>Method - `actionCategory.ACTION_VERB` - type: FUNCTION [#](#actionCategory-ACTION)
+Each passed in action verb in the actions array becomes a key within the actionCategory that maps to a method which can be invoked to dispatch the corresponding action.
+
+```
+    actionCategory.add(actionBody); //OR
+    actionCategory.update(actionBody); //ETC
+```
+
+#####<a id="ACTION-actionBody"></a>Parameter - `actionBody` - type: OBJECT - optional [#](#ACTION-actionBody)
+Action body which will be passed in to any listeners on this action.
+
+```
+    var newTodo = {
+      text: 'the actionCategory.add method will invoke the storeToRegister.onAdd '
+      + 'callback and will pass this newTodo object into it'
+    };
+    actionCategory.add(newTodo);
+```
+
+#####<a id="ACTION-type"></a>Property - `ACTION_VERB.type` - type: STRING
+This property is attached to each action verb method and represents the full name of the action (category + '_' + action).  TuxedoJS uses this to find the callbacks that are listening to the dispatched action.  You should not normally need access to this property.
+
+```
+    actionCategory.add.type === 'todos_add';
+```
+
+####<a id="actionCategory-before"></a>Method - `actionCategory.before` - type: FUNCTION [#](#actionCategory-before)
+Accepts an action verb string or array of action verbs and a callback, invokes the callback before dispatching the action when the action method is invoked.  The major use case for the `before` callback is to allow actions to invoke async functions (such as Ajax requests) and then dispatch the results of those requests.
+
+```
+    actionCategory.before(actionVerb, callbackToInvokeBeforeDispatching);
+```
+
+#####<a id="before-actionVerb"></a>Parameter - `actionVerb` - type: STRING or ARRAY of STRINGs - required [#](#before-actionVerb)
+Action (or array of actions) which will invoke the passed in callback before dispatching their corresponding actions.
+
+```
+    actionVerb = 'add'; //OR
+    actionVerb = ['add', 'remove'];
+```
+
+#####<a id="before-callbackToInvokeBeforeDispatching"></a>Parameter - `callbackToInvokeBeforeDispatching` - type: FUNCTION - required [#](#before-callbackToInvokeBeforeDispatching)
+Function to be invoked before the action to dispatch. Multiple `callbackToInvokeBeforeDispatching` callbacks can be registered to an action via the `before` method and they will be invoked in reverse order of how they are registered.  The last function to invoke will dispatch the action.  This chaining occurs by having each callback invoke the next callback in the series and, optionally, pass in an `actionBody` to it. This syntax allows for asynchronous callback chaining. The inputs to the callback are described below
```

Comment:
The third sentence might read better as "The last function to be invoked will dispatch the action."